### ISSUE TITLE
[RFC] Rename the Python binary extension name to be something like `C`

### ISSUE THREAD

**Issue by ProfFan** [borglab/gtsam#1243]
See #1242 

I think PyTorch does this, the extension is named differently than the package name so one cannot ever import the inner module inadvertently.

**Comment by varunagrawal** [comment]
Yeah I tried looking into this but I couldn't figure it out then. If you get some insight, we can land this pretty quickly.
